notebooks/3_SinglePlotDiagnostics/1_Ins_SingleModelVisualDiagnostics.ipynb

```diff
@@ -0,0 +1,827 @@
+    "\n",
+    "This is not to say finite MCMC chains are always wrong, but it does mean that when using MCMC inference methods it is prudent to check the results diagnostic methods. In this tutorial we'll cover both numerical and visual methods\n",
+    "\n",
+    "### Desirable characteristics in MCMC chains\n",
+    "Bayesian MCMC modelers want *a large number of samples from well mixed chains*. Typically this indicates that *the chains are stationary and have converged*. We'll illustrate the meaning and intuition behind these statements below."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## A story of two Inference Runs\n",
+    "In demonstrating what good and \"bad\" diagnostics look like we will be reusing our Metropolis Hasting sampler. \n",
+    "\n",
+    "Recall that our data has been generated from the following distribution\n",
+    "$$ y \\sim N(30,1) $$\n",
+    "\n",
+    "\n",
+    "**Insert Model Here**\n",
+    "\n",
+    "For our small simulated model, by mathematical intuition, we know that $mu \\sim 30-20 \\sim 10$ but let's see if we can get there using Inference\n",
+    "\n",
+    "Let's generate some data first"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data = stats.norm.rvs(loc=30, scale=1, size=1000).flatten()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Visual Diagnostics\n",
+    "### Traceplot"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Traceplots help us \"trace\" the steps of the MCMC chain. Things to look for in traceplots are \n",
+    "* Mixing of chains\n",
+    "* No long regions of monoticity \n",
+    "\n",
+    "This indicates that the MCMC chain is able to readily sampling the posterior distribution and provide some confidence that regions of the posterior are not being excluded. Traceplots can help diagnose problems such as \n",
+    "\n",
+    "* Multimodal posteriors\n",
+    "* Bad intialization\n",
+    "* Difficult sampling topologies (such as Neal's funnel)\n",
+    "\n",
+    "Typically you'll hear that traceplots should look like \"fuzzy\" caterpillars. In ArviZ `plot_trace` returns a traceplot for each variable on the right column, and a posterior plot of the samples on the left."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Example 1: Bad initialization\n",
+    "Recall that MCMC is very sensitive to starting points of each chain, similar to gradient descent or random weights in Neural Nets. To illusrate example let's \"randomly\" pick -50 and 60 as our starting points for each chain."
```

Comment:
Super true (unless it is initialized outside of the support of the distribution, or it runs into numerical accuracy problems). 

Maybe the more correct, nuanced-but-not-too-nuanced way of saying this is the way Betancourt does: MCMC goes in 3 stages - 
1. finding the typical set
2. initial exploration
3. more exploration

In a perfect world, you want to spend most of your computing time on step 3 (and all of your parameter tuning on step 2). Bad initialization can make step 1 take longer, and maybe you accidentally do your parameter tuning out on the tails, which make step 3 less efficient.

Specifically, both MH and HMC use a fixed step size, so if you start distance 60 from the mean, and set the step size to 0.1 (...and don't tune parameters), it will take at least 600 steps to get close to the mean.

Reply:
So it is sensitive to starting points?

Reply:
A proper MCMC sample should not be sensitive to the starting point. That's why you can use it as a diagnostic. If the starting point makes a difference then something is wrong.